tests/test_surface/test_file_io.py

```diff
@@ -0,0 +1,144 @@
+import pytest
+import deprecation
+
+from hypothesis import given, settings
+from hypothesis import strategies as st
+import numpy as np
+
+from xtgeo import RegularSurface, Cube
+import xtgeo
+
+
+@pytest.mark.usefixtures("setup_tmpdir")
+@pytest.mark.parametrize("engine", ["cxtgeo", "python"])
+@pytest.mark.parametrize(
+    "fformat",
+    [
+        "irap_binary",
+        "irap_ascii",
+        "zmap_ascii",
+        # "ijxyz",  # This fails horribly
+        "petromod",
+        "zmap",
+        "xtgregsurf",
+    ],
+)
+@pytest.mark.parametrize(
+    "input_val, expected_result",
+    [
+        (1, [[1.0, 1.0], [1.0, 1.0]]),
+        ([1, 2, 3, 4], [[1.0, 2.0], [3.0, 4.0]]),
+        (np.zeros((2, 2)), [[0.0, 0.0], [0.0, 0.0]]),
+        (np.ma.zeros((2, 2)), [[0.0, 0.0], [0.0, 0.0]]),
+        (np.ma.ones((2, 2)), [[1.0, 1.0], [1.0, 1.0]]),
+    ],
+)
+def test_simple_io(input_val, expected_result, fformat, engine):
```

Comment:
How about setting this test up with the product of the format? mix formats, using one for `to_file` and the other for `from_file`?

Reply:
What do you mean here exactly? I cant write a `roff` file and read an `ascii` file?

Reply:
Brain fart.. 

A possible thing to do would be:

```
surf = read_roff("surf.roff")
surf.to_ascii("surf.ascii")
reloaded_surf = read_ascii("surf.ascii")
reloaded_surf.to_roof("surf_by_ascii.roof")
```

And then assert that `surf.roff` and `surf_by_ascii.roff` were identical. But we might not even guarantee that they are identical, for instance if some metadata doesn't carry over, or if we ignore comments.. 

And besides, the test you've made already would cover these steps individually.. So yeah, it's a poor suggestion